profiling/test_import_and_query.py

```diff
@@ -0,0 +1,190 @@
+                data_object=data_object,
+                vector=vector,
+                class_name=name,
+                uuid=uuid.UUID(int=i),
+            )
+
+    print(f"V3: Finished writing {len(vectors)} records in {time.time()-start}s")
+
+
+def query_v4(
+    collection: Collection, vectors: List[List[float]], neighbours: List[List[int]], ef: int
+) -> None:
+    collection.config.update(vector_index_config=wvc.config.Reconfigure.VectorIndex.hnsw(ef=ef))
+    start = time.time()
+    recall = 0.0
+
+    for i, vec in enumerate(vectors):
+        objs = collection.query.near_vector(
+            near_vector=list(vec), limit=LIMIT, return_properties=[]
+        ).objects
+        ideal_neighbors = set(neighbours[i][:LIMIT])
+        res_ids = [obj.uuid.int for obj in objs]
+        recall += len(ideal_neighbors.intersection(res_ids)) / LIMIT
+
+    print(
+        f"V4: Querying {len(vectors)} records with ef {ef} in {time.time()-start}s with recall {recall/len(vectors)}"
+    )
+
+
+def query_v3(
+    collection: Collection,
+    client: weaviate.Client,
+    vectors: List[List[float]],
+    neighbours: List[List[int]],
+    ef: int,
+) -> None:
+    collection.config.update(vector_index_config=wvc.config.Reconfigure.VectorIndex.hnsw(ef=ef))
+    start = time.time()
+    recall = 0.0
+
+    for i, vec in enumerate(vectors):
+        res = (
+            client.query.get(collection.name, ["i _additional{id}"])
+            .with_near_vector(
+                {
+                    "vector": vec,
+                }
+            )
+            .with_limit(LIMIT)
+            .do()
+        )
+        res_ids = [
+            uuid.UUID(res["_additional"]["id"]).int for res in res["data"]["Get"][collection.name]
+        ]
+        ideal_neighbors = set(neighbours[i][:LIMIT])
+
+        recall += len(ideal_neighbors.intersection(res_ids)) / LIMIT
+
+    print(
+        f"V3: Querying {len(vectors)} records with ef {ef} in {time.time()-start}s with recall {recall/len(vectors)}"
+    )
+
+
+def run_v4(file: str, name: str, efc: int, m: int) -> None:
+    sift_file = get_file_path(file)
+
+    f = h5py.File(sift_file)
+    vectors_import = f["train"]
+    vectors_test = f["test"]
+    ideal_neighbors = f["neighbors"]
+
+    client = weaviate.connect_to_local()
+    collection = create_schema(client, name, efc, m, 1, "l2-squared")
+    load_records_v4(collection, vectors_import)
+    for ef in EF_VALUES:
+        query_v4(collection, vectors_test, ideal_neighbors, ef)
+
+
+def run_v3(file: str, name: str, efc: int, m: int) -> None:
+    sift_file = get_file_path(file)
+
+    f = h5py.File(sift_file)
+    vectors_import = f["train"]
+    vectors_test = f["test"]
+    ideal_neighbors = f["neighbors"]
+
+    client = weaviate.Client(url="http://localhost:8080")
+
+    # use v4 client to create schema to avoid duplicate code
+    clientv4 = weaviate.connect_to_local()
+    collection = create_schema(clientv4, name, 128, 32, 1, "l2-squared")
```

Comment:
`efc` and `m` appear hard-coded here, was this intentional?

Reply:
yes, parameters are from here: https://weaviate.io/developers/weaviate/benchmarks/ann#sift1m-1m-128d-vectors-l2-distance

Reply:
Does this call: https://github.com/weaviate/weaviate-python-client/blob/30afbb9af7fa7499d24cbcc5a04a3ab2b5f021a3/profiling/test_import_and_query.py#L182 need changing then since its params aren't being respected?

Reply:
oh now I see what you mean! Yes, that needs fixing